docs/user_guide/hyperopt.md

```diff
@@ -27,6 +30,50 @@ hyperopt:
   metric: loss
 ```
 
+## Default Feature - Level Parameters
+
+In addition to defining hyperopt parameters for individual input or output features (like the `title` feature
+in the example above), feature level parameters can be specified for entire feature types. These parameters will
+follow the same convention as the `defaults` section of the Ludwig config. This is particularly helpful in cases
+where a dataset has a large number of features and you don't want to define parameters for each feature individually.
+
+Feature level parameters are defined using the following keywords in order separated by the `.` delimiter:
+
+- `defaults`: The defaults keyword used to indicate a feature-level parameter.
+- `feature_type`: Any input or output feature type in the current Ludwig config.
+- `subsection`: One of `preprocessing`, `encoder`, `decoder` or `loss`.
+- `parameter`: A valid parameter belonging to the `subsection`. For e.g., `most_common` is a valid parameter for the
+`preprocessing` section for `text` feature type.
+
+For each hyperopt trial, a value will be sampled from the feature level parameter space and applied to either input features
+(`preprocessing` or `encoder` related parameters) or output features (`decoder` or `loss` related parameters) of that feature type
+as long as they use the default encoder (for input features) or default decoder (for output features). Additionally, parameters
+defined for individual features (like `title.num_filters`) will take precedence over feature level parameters
+(like `defaults.text.encoder.num_filters`) if they share the same type and parameter.
+
+```yaml title="config.yaml"
+hyperopt:
+  parameters:
+    title.num_filters:
+      space: choice
+      categories: [128, 256, 512]
+    defaults.text.encoder.num_filters:
+      space: choice
+      categories: [128, 256, 512]
+    defaults.category.decoder.reduce_input:
+      space: choice
+      categories: ['mean', 'sum', 'max']
+  goal: minimize
+  metric: loss
+```
+
+In this example, there are two feature level parameters defined:
+
+- `defaults.text.encoder.num_filters`: This will apply the sampled `num_filters` value to all input text features
```

Comment:
Can you add a note here to reiterate that this won't update the `num_filters` search space for the `title` feature?